libsolidity/formal/SymbolicVariables.cpp

```diff
@@ -37,16 +37,32 @@ SymbolicVariable::SymbolicVariable(
 {
 }
 
+smt::Expression SymbolicVariable::currentValue() const
+{
+	return valueAtIndex(m_ssa->index());
+}
+
 string SymbolicVariable::currentName() const
 {
 	return uniqueSymbol(m_ssa->index());
 }
 
+smt::Expression SymbolicVariable::valueAtIndex(int _index) const
+{
+	return m_interface.newVariable(uniqueSymbol(_index), smtSort(*m_type));
+}
+
 string SymbolicVariable::uniqueSymbol(unsigned _index) const
 {
 	return m_uniqueName + "_" + to_string(_index);
 }
 
+smt::Expression SymbolicVariable::increaseIndex()
+{
+	++(*m_ssa);
```

Comment:
@leonardoalt  actually, LGTM, but maybe what about solAssert()'ing in `increaseIndex()` to make sure you your index doesn't get out of bounds?

Reply:
Could do it, but don't you think that'd be a bit of an overkill? It's an `unsigned`, if we have that many assignments to a variable we'd probably run into other problems ;)